src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ClientBaseTests.cs

```diff
@@ -174,6 +174,48 @@ public static void ClientBaseOfT_Sync_RoundTrip_Call_Using_NetTcpTransport()
         }
     }
 
+    [Fact]
+    [OuterLoop]
+    public static void OperationContextScope_HttpRequestCustomMessageHeader_RoundTrip_Verify()
+    {
+        BasicHttpBinding binding = new BasicHttpBinding();
+        MyClientBase<IWcfService> client = new MyClientBase<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
+        IWcfService serviceProxy = client.ChannelFactory.CreateChannel();
+
+        string customHeaderName = "OperationContextScopeCustomHeader";
+        string customHeaderNS = "http://Microsoft.WCF.OperationContextScope_HttpRequestCustomMessageHeader_RoundTrip_Verify";
+        string customHeaderValue = "CustomHappyValue";
+        try
+        {
+            using (OperationContextScope scope = new OperationContextScope((IContextChannel)serviceProxy))
+            {
+                MessageHeader header
+                  = MessageHeader.CreateHeader(
+                  customHeaderName,
+                  customHeaderNS,
+                  customHeaderValue
+                  );
+                OperationContext.Current.OutgoingMessageHeaders.Add(header);
+
+                string result = serviceProxy.GetRequestCustomHeader(customHeaderName, customHeaderNS);
+
```

Comment:
The server currently find the value of the specified custom header from it's incoming headers , it returns null if not found. The the test fails if it gets null result.

Return all custom headers could make the test validation more obvious. I'll make the change.